boardswarm-cli/src/ui.rs

```diff
@@ -206,7 +238,11 @@ pub async fn run_ui(
                         Input::Up => { terminal.scroll_up(); },
                         Input::Down => { terminal.scroll_down(); },
                         Input::ScrollReset => { terminal.scroll_reset(); },
-                        _ => (),
+                        Input::PowerOn => { terminal.set_power_on(); },
+                        Input::PowerOff => { terminal.set_power_off(); },
+                        Input::PowerReset => { terminal.set_power_reset(); },
+                        Input::ShowState => { terminal.show_state(); },
```

Comment:
Really the device mode should be coming from the actual device mode; not what a ui has sent ..  You can listen to the device-info stream ; See how e.g. the implementation of `boardswarm-cli device X info --follow`

Reply:
Would the right place to listen to device info stream be https://github.com/boardswarm/boardswarm/blob/7c04a7c8d41ba0fbfdb2892926157c7c93a3b1f1/boardswarm-cli/src/ui.rs#L196?

Reply:
I'd probably do it in a seperate task and only message the main thread when there is a mode update specificially such that it can trigger a next frame.